File: src/main/java/PreProcessing.java
import java.io.File;
import java.io.IOException;
import java.net.URL;

public class PreProcessing {
    public static String preprocesses(String path, String startString){
        URL url;
        url = PreProcessing.class.getResource(path);

        if (url == null) return "INVALID PATH!";

        File directoryPath = new File(url.getPath());
        File[] filesList = directoryPath.listFiles();
        try {
            InvertedIndex index = new InvertedIndex();
            if (filesList == null) return "INVALID FILE LIST!";
            else {
                addFilesToIndexFiles(filesList, index);
                new InputScannerView(index, startString);
                return "end";
            }
        } catch (Exception e) {
            return "an error has happened!";
        }

    }

    private static void addFilesToIndexFiles(File[] filesList, InvertedIndex index) throws IOException {
        for (File file : filesList) {
            index.indexFile(file);
        }
    }
}

File: ConsoleApp1/PreProcessing.cs
using System;
using System.IO;

namespace ConsoleApp1
{
    public class PreProcessing
    {
        static string invalidPathString = "INVALID PATH!";

        public static string Preprocesses()
        {
            string[] files = Directory.GetFiles(@"..\..\..\..\TestProject1\");

            try
            {
                IInvertedIndex index = new InvertedIndex();

                addFilesToIndexFiles(files, index);
                new InputScannerView(index);
                return "end";
            }
            catch (Exception e)
            {
                return "an error has happened!";
            }
        }

        private static void addFilesToIndexFiles(string[] filesList, IInvertedIndex index)
        {
            index.IndexFile(filesList);
        }
    }
}